File: shuttle_tablet/lib/features/trips/domain/entities/trip_passenger_email_log.dart
import 'package:equatable/equatable.dart';

class TripPassengerEmailLog extends Equatable {
  final String id;
  final String tripPassengerId;
  final String recipientEmail;
  final String direction;
  final DateTime sentAt;
  final bool isTest;

  const TripPassengerEmailLog({
    required this.id,
    required this.tripPassengerId,
    required this.recipientEmail,
    required this.direction,
    required this.sentAt,
    required this.isTest,
  });

  @override
  List<Object?> get props =>
      [id, tripPassengerId, recipientEmail, direction, sentAt, isTest];
}

File: shuttle-api/src/ShuttleApi.Domain/Trips/TripPassengerEmailLog.cs
using ShuttleApi.Domain.Common;

namespace ShuttleApi.Domain.Trips;

public sealed class TripPassengerEmailLog : Entity<Guid>
{
    private TripPassengerEmailLog() { }

    public static TripPassengerEmailLog Record(
        Guid passengerId, string recipientEmail, string direction, bool isTest) =>
        new()
        {
            Id = Guid.NewGuid(),
            TripPassengerId = passengerId,
            RecipientEmail = recipientEmail,
            Direction = direction,
            SentAt = DateTime.UtcNow,
            IsTest = isTest
        };

    public Guid TripPassengerId { get; private set; }
    public string RecipientEmail { get; private set; } = default!;
    public string Direction { get; private set; } = default!;
    public DateTime SentAt { get; private set; }
    public bool IsTest { get; private set; }
}
